oauthproxy.go

```diff
@@ -108,6 +109,8 @@ type OAuthProxy struct {
 	serveMux          *mux.Router
 	redirectValidator redirect.Validator
 	appDirector       redirect.AppDirector
+
+	shouldEncodeState bool
```

Comment:
Please don't use weak language/words. `Should` is a weak word because it doesn't guarantee that something is truly done. 

```suggestion
	encodeState bool
```